Update docs with changelog guidelines (#15537)
DataDog/integrations-core: docs/developer/guidelines/pr.md

...
```

> Unless you have applied the `changelog/no-changelog` label, you must also edit the `CHANGELOG.md` file in 

Reviewer: ```suggestion
If you don't apply the `changelog/no-changelog` label, you must edit the `CHANGELOG.md` file in 
```

the integration you have modified. Below are the guidelines for the correct formatting for changelogs: 

### Spacing

> * There should be one empty line between all text of different format. This means that there should be an 
> empty line between all of the following sections of text:
>   * Changelog file header
>   * Unreleased header
>   * Version / Date header
>   * Change type (ex: fixed, added, etc)
>   * Specific descriptions of changes (note: within this section itself, there should NOT be new lines between bullets)

Reviewer: ```suggestion
* There should be a blank line between each section. This means that there should be a line between the following sections of text:
  * Changelog file header
  * Unreleased header
  * Version / Date header
  * Change type (ex: fixed, added, etc)
  * Specific descriptions of changes (**Note**: Within this section, there should **not** be new lines between bullet points,)
```

> * `Extra spacing on line {line number}`: you have an extra empty line on the line referenced in the error. 
> There should be only one line of spacing!
> * `Missing spacing on line {line number}`: you need to add an empty line to the referenced lie

Reviewer: ```suggestion
* `Extra spacing on line {line number}`: There is an extra blank line on the line referenced in the error. 
* `Missing spacing on line {line number}`: Add an empty line above or below the referenced line.
```

### Integration Version Header

> * The header for an integration version should be in the following fashion: `version number / YYYY-MM-DD / Agent Version Number`. 

Reviewer: ```suggestion
* The header for an integration version should be in the following format: `version number / YYYY-MM-DD / Agent Version Number`. 
```

The Agent version number is not necessary, but a valid version number and date are required. The first header after the 
file's title can be `Unreleased`. The content under this section is the same as any other. 
* `Version is formatted incorrectly on line {line number}`: The version you inputted is not a valid version, or there is 
no / separator between the version and date in your header.
* `Date is formatted incorrectly on line {line number}`: The date must be formatted as YYYY-MM-DD, with no spaces in between. 

### Changelog Content

> * The changelog content is broken down by the different type of change that was made. For changelog types, we adhere 
> to those defined by [Keep a Changelog][keepachangelog-types]. For each different version, they must be written in the 
> following order, by priority:

Reviewer: ```suggestion
* The changelog content is broken down by the type of change that was made. For changelog types, we adhere to those defined by [Keep a Changelog][keepachangelog-types]. For each new version, they must be written in the following order, by priority:
```

Reviewer: I think the anchor link here is broken.

Author: https://datadoghq.dev/integrations-core/guidelines/pr/ on here the existing link seems to work so i just kept it as is

> * The changelog content is broken down by the different type of change that was made. For changelog types, we adhere 
> to those defined by [Keep a Changelog][keepachangelog-types]. For each different version, they must be written in the 
> following order, by priority:
>   * ***Removed***: for now removed features.
>   * ***Changed***: for changes in existing functionality.
>   * ***Security***: in case of vulnerabilities.
>   * ***Deprecated***: for soon-to-be removed features.
>   * ***Added***: for new features or any non-trivial refactors.
>   * ***Fixed***: for any bug fixes.

Reviewer: ```suggestion
* The changelog content is broken down by the type of change that was made. For changelog types, we adhere to those defined by [Keep a Changelog][keepachangelog-types]. For each version, they must be written in the following order, listed by priority:
  * ***Removed*** for features that have been removed.
  * ***Changed*** for changes to existing functionality.
  * ***Security*** in case of vulnerabilities.
  * ***Deprecated*** for soon-to-be removed features.
  * ***Added*** for new features or any non-trivial refactors.
  * ***Fixed*** for any bug fixes.
```

>   * ***Removed***: for now removed features.
>   * ***Changed***: for changes in existing functionality.
>   * ***Security***: in case of vulnerabilities.
>   * ***Deprecated***: for soon-to-be removed features.
>   * ***Added***: for new features or any non-trivial refactors.
>   * ***Fixed***: for any bug fixes.

Reviewer: ```suggestion
  * ***Removed*** for features that have been removed.
  * ***Changed*** for changes to existing functionality.
  * ***Security*** in case of vulnerabilities.
  * ***Deprecated*** for soon-to-be removed features.
  * ***Added*** for new features or any non-trivial refactors.
  * ***Fixed*** for any bug fixes.
```

* The changelog header must be capitalized and written in this format: `***HEADER***:` It should be bolded and italicized!

> * `Changelog type is incorrect on line {line count}`: The changelog header on that line is not one of the 6 valid changelog types

Reviewer: ```suggestion
* `Changelog type is incorrect on line {line count}`: The changelog header on that line is not one of the six valid changelog types.
```
